Core/Domibus-MSH/src/main/java/eu/domibus/core/message/UserMessageDefaultService.java

```diff
@@ -858,6 +861,14 @@ protected byte[] messageToBytes(UserMessage userMessage, List<PartInfo> partInfo
     }
 
     protected String getPayloadName(PartInfo info) {
+        if (BooleanUtils.isTrue(domibusPropertyProvider.getBooleanProperty(DOMIBUS_PAYLOAD_BUSINESS_CONTENT_ATTACHMENT_ENABLED))) {
+            for (PartProperty property : info.getPartProperties()) {
+                if (DOMIBUS_PAYLOAD_BUSINESS_CONTENT_ATTACHMENT.equalsIgnoreCase(property.getName()) && property.getValue() != null) {
```

Comment:
Instead of checking for `null`:

```
property.getValue() != null
```

Would it be better to check for [non-empty](https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html) (not `null` or empty) value since an empty string may cause an exception later in the process:

```
!StringUtils.isEmpty(property.getValue())
```

Reply:
Replaced this null check with isEmpty()